gadgets/trace_tcpconnect/program.bpf.c

```diff
@@ -372,6 +422,62 @@ int BPF_KRETPROBE(ig_inet_stream_connect_ret, int ret)
 	return exit_tcp_connect(ctx, ret);
 }
 
+SEC("kretprobe/tcp_v4_connect")
+int BPF_KPROBE(ig_tcp_v4_connect, int ret)
+{
+	__u64 pid_tgid = bpf_get_current_pid_tgid();
+	__u32 tid = pid_tgid;
+	struct sock **skpp;
+	struct sock *sk;
+	struct src_dst src_dst_entry;
+	unsigned short family;
+
+	skpp = bpf_map_lookup_elem(&sockets_per_process, &tid);
+	if (!skpp)
+		return 0;
+	sk = *skpp;
+
+	family = BPF_CORE_READ(sk, __sk_common.skc_family);
+	if (family == AF_INET) {
+		read_l4endpoints_from_sock_v4(&src_dst_entry.src,
+					      &src_dst_entry.dst, sk);
+	} else {
+		read_l4endpoints_from_sock_v6(&src_dst_entry.src,
+					      &src_dst_entry.dst, sk);
+	}
+
+	bpf_map_update_elem(&src_dst_per_process, &tid, &src_dst_entry, 0);
+	return 0;
+}
+
+SEC("kretprobe/tcp_v6_connect")
```

Comment:
IMO, the logic became quite complex. I suggest creating a mermaid diagram as we did for [fsnotify](https://www.inspektor-gadget.io/docs/latest/gadgets/fsnotify#architecture) where we describe what we do at each hook. In particular, how programs interact with maps. WDYT?

Reply:
It'd be highly appreciated 🙂

Reply:
Could you please take a look. Since I am currently too familiar with the topic I want to make sure its understandable for everyone